src/viewer.js

```diff
@@ -0,0 +1,135 @@
+import { getElement } from './utils';
+import {
+  getImageDataByLocation,
+  getImageDataById
+} from './services/panorama-api/panorama-api';
+import { loadScene, initialize } from './services/marzipano/marzipano';
+import { radiansToDegrees } from './services/angle-conversion/angle-conversion';
+
+const viewerOpts = {
+  controls: {
+    mouseViewMode: 'drag'    // drag|qtvr
+  },
+  stageType: null,
+  stage: {
+    preserveDrawingBuffer: true
+  }
+};
+
+class PanoViewer {
+
+  constructor(elementId) {
+    this.pov = {
+      fov: 80,
+      yaw: 0,
+      pitch: 0
+    }
+    this.tags = null;
+    this.events = null;
+
+    const panoElement = getElement(elementId);
+    if (!panoElement) {
+      return Error('No dom element available');
+    }
+    this.viewer = initialize(panoElement, viewerOpts);
+
+    this._loadScene = this._loadScene.bind(this);
+    this._updatePanorama = this._updatePanorama.bind(this);
+    this._updateLocation = this._updateLocation.bind(this);
+    this._bindEvents = this._bindEvents.bind(this);
+  }
+
+  _loadScene(data) {
+    try {
+      const image = data.image;
+      const hotspots = data.hotspots;
+      const { yaw, pitch, fov } = this.pov;
+      const onClick = this._updatePanorama;
+      const scene = loadScene(this.viewer, onClick, image, yaw, pitch, fov, hotspots);
+
+      const view = scene.view();
+      this._updateLocation(data);
+      this._bindEvents(view);
+    } catch (e) {
+      console.error(`Error Loading scene: ${e}`);
+    }
+  }
+
+  _updateLocation(data) {
+    const location = {
+      date: data.date,
+      lat: data.location[0],
+      lon: data.location[1]
+    };
+    if (this.events && this.events.location) {
+      this.events.location(location);
+    }
+  }
+  /**
+   *
+   * @param {float} lat - Latitude
+   * @param {float} lon - Longtitude
+   * @param {float} [yaw] - initial yaw
+   * @param {float} [pitch] - initial pitch
+   * @param {float} [fov] - Field of vision
+   *
+   * This is the main api interaction for loading the panorama view.
+   * After this has been called navigating within the panorama via hotspots
+   * is handled by _updatePanorama
+   */
+  loadPanorama(lat, lon, tags, yaw, pitch, fov) {
+    // Updating POV if needed
+    this.pov.fov = fov || this.pov.fov;
```

Comment:
Please rewrite this to use parameter defaults

Reply:
done